backend/handlers/institution_parent_request_handler.py

```diff
@@ -56,5 +58,11 @@ def delete(self, user, request_key):
         request.change_status('rejected')
         request.put()
 
+        parent_institution = request.institution_requested_key.get()
+
+        if request.institution_key in parent_institution.children_institutions:
```

Comment:
I suggest you to create a remove_child method, the opposite of add_child, to keep pattern.